android/src/main/java/com/mvcpscrollviewmanager/MvcpScrollViewManagerModule.java

```diff
@@ -117,7 +160,11 @@ public void disableMaintainVisibleContentPosition(int key, Promise promise) {
       if (key >= 0) {
         final UIManagerModule uiManagerModule = this.reactContext.getNativeModule(UIManagerModule.class);
         uiManagerModule.removeUIManagerListener(uiManagerModuleListeners.remove(key));
-        uiManagerModule.getUIImplementation().removeLayoutUpdateListener();
+        // only remove the layout listener when there are no more ui module listeners
```

Comment:
Only remove the layout listener when there are no more module listeners, otherwise, when having multiple instances of Flatlist, the module will not get new updates since the listener will be detached.